src/OpenTelemetry/Logs/ILogger/OpenTelemetryLogger.cs

```diff
@@ -111,7 +110,7 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     public bool IsEnabled(LogLevel logLevel)
     {
-        return logLevel != LogLevel.None;
+        return logLevel != LogLevel.None && !Sdk.SuppressInstrumentation;
```

Comment:
it was never clear to me why the LogLevel.None check was needed.. If the LogLevel.None is used, the ILogger infra should not be even sending the log to this provider at all (maybe thats just my interpretation!)

Reply:
@cijothomas 

It does appear somewhat standard: https://github.com/dotnet/runtime/blob/703f6e846027681a052970929989bd36e81636cd/src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLogger.cs#L73-L76

But you are correct the `LoggerFactory` also does a fast-check up front: https://github.com/dotnet/runtime/blob/703f6e846027681a052970929989bd36e81636cd/src/libraries/Microsoft.Extensions.Logging/src/LoggerInformation.cs#L32-L35

My guess would be it is because someone might swap out the default `ILoggerFactory` with their own implementation which might not have the logic?

Reply:
Yes it appears standard in the sense - every logging provider I have seen has that logic! I wasn't expecting that providers have to do that job, instead handled by ILogger itself.!